Sources/XCTest/XCTestCase.swift

```diff
@@ -178,3 +178,20 @@ private func test<T: XCTestCase>(_ testFunc: (T) -> () throws -> Void) -> (XCTes
         try testFunc(testCase)()
     }
 }
+
+/// Wrapper function for the non-throwing variant of tests.
+/// - seealso: `XCTMain`
+public func testCase<T: XCTestCase>(_ allTests: [(String, (T) -> () -> Void)]) -> XCTestCaseEntry {
+    let tests: [(String, (XCTestCase) throws -> Void)] = allTests.map { ($0.0, test($0.1)) }
+    return (T.self, tests)
+}
+
+private func test<T: XCTestCase>(_ testFunc: (T) -> () -> Void) -> (XCTestCase) throws -> Void {
```

Comment:
I don't think it's necessary to add this overload, because the type conversion should allow the existing `test(_:)` function to be used (no tuples involved in the argument type).